### PR TITLE
Updated user endpoints

### PR DESCRIPTION
**Pull request by k-stewart**
Basically [Vivian's branch](https://github.com/EFForg/eff_fab/pull/87), but rebased with master, some new stuff TechOps asked for, and fancy docs.

### CODE REVIEW

File: doc/api.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+Our FABulous API Docs
```
Comment (vbrown608) [inline doc/api.md]: :100:

File: README.md [hunk 2]
```diff
@@ -50,10 +50,23 @@ Reminders and report_on_aftermath notifications are sent over email via `rake ma
...
+-------------------
+
+### From within the app
+
+1. Log in as a user with the admin role.
+2. Click "Administer" in the nav bar and click "Add new user".
+
+### From the API
+
+Like the app, the API uses basic auth for authentication. Along with your request, you must pass basic auth credentials associated with a user who has the admin role. In other words if the basic auth name is [NAME], there must be user with admin credentials and the email address [NAME]@eff.org.
```
Comment (vbrown608) [inline README.md]: Keeping API access separate from basic auth sounds great to me! We should update this.

**Comment by k-stewart** [comment]
> There's a section in the docs called "without authentication, fails". This makes sense as a spec but seems like a confusing thing to have in docs. Is it possible to mark it for exclusion?

It's definitely possible to exclude it (maybe the simplest way is to move that spec out of the acceptance tests and into the requests), but I wanted to keep it in the docs so that TechOps knew what to expect if their authentication failed.

**Comment by k-stewart** [comment]
> The link to "Offboarding previous employees" in the sidebar doesn't take me anywhere.

Hmm, interesting.  I've attached a gif of navigating through the sidebar links locally.  It works about as I expect it to; one caveat is that if I scroll to the first example in a section and then click the section link, nothing happens, I guess because I'm already at the anchor.  Is it different for you?
doc_links

**Comment by k-stewart** [comment]
Sounds good to me.  I'll reach out to them and also to Ben, who expressed interest in the docs.

**Comment by k-stewart** [comment]
Oh alas, Rocket is on vacation.  How would you feel about merging this, and then making changes as Rocket and Ben request?  If that seems useful, what changes would make this mergeworthy?

**Comment by vbrown608** [comment]
Happy to merge and get possible updates later :)

**Pull request merged**